io/disk/Avago_storage_adapter/avago9361_vd.py

```diff
@@ -150,6 +156,99 @@ def test_online_offline(self):
+                                                            self.hotspare)
+        self.check_pass(cmd, "Failed to delete hotsparedrive")
+        cmd = "./storcli64 /c%d/%s add hotsparedrive DGs=0" % (self.controller,
+                                                               self.hotspare)
+        self.check_pass(cmd, "Failed to set DHS")
+        self.set_online_offline('offline')
+        self.rebuild_progress(self.hotspare)
+        self.copyback_operation()
+        self.vd_delete()
+
+    def copyback_state(self, state):
+
+        """
+        Helper function to run copyback test
+        """
+        if state == 'start':
+            cmd = "./storcli64 /c%d/%s start copyback target=%s" \
+                   % (self.controller, self.hotspare, self.copyback)
+        else:
+            cmd = "./storcli64 /c%d/%s %s copyback" % (self.controller,
+                                                       self.on_off, state)
+        self.check_pass(cmd, "Failed to start copyback")
+
+    def copyback_operation(self):
+
+        """
+        Function to run copyback operations
+        """
+        for state in ['start', 'stop', 'start', 'pause', 'resume']:
+            self.copyback_state(state)
+            cmd = "./storcli64 /c%d/%s show copyback" % (self.controller,
+                                                         self.on_off)
+            self.showprogress(cmd)
+        while self.showprogress(cmd) is True:
+            time.sleep(30)
+
+    def test_rebuild(self):
+
+        """
+        Function to handle rebuild scenario
+        """
+        self.vd_create('WB', 'nora', 'direct', 256)
+        self.set_online_offline('offline')
+        self.rebuild_operations()
+        self.vd_delete()
+
+    def rebuild_operations(self):
+
+        """
+        Helper function of all types of rebuild operations
+        """
+        for state in ['start', 'stop', 'start', 'pause', 'resume']:
+            self.rebuild_state(state)
+            cmd = "./storcli64 /c%d/%s show rebuild" % (self.controller,
+                                                        self.on_off)
+            self.showprogress(cmd)
+        while self.showprogress(cmd) is True:
+            time.sleep(30)
+
+    def rebuild_progress(self, disk):
+
+        """
+        Helper function to see the progress of rebuild operation
+        """
+        cmd = "./storcli64 /c%d/%s show rebuild" % (self.controller,
+                                                    disk)
+        self.showprogress(cmd)
+        while self.showprogress(cmd) is True:
+            time.sleep(30)
+
+    def rebuild_state(self, state):
+
+        """
+        Helper function for all CC operations
+        """
+        cmd = "./storcli64 /c%d/%s %s rebuild" % (self.controller,
+                                                  self.on_off, state)
+        self.check_pass(cmd, "Failed to %s Rebuild" % state)
+        time.sleep(10)
```

Comment:
@balamuruhans Generally, I agree. But since these are IO operations, they need some time before starting any other operation.
@vrbagalkote Pls check if all such sleeps are valid.